src/main/java/com/hartwig/platinum/kubernetes/pipeline/PipelineJob.java

```diff
@@ -49,7 +49,9 @@ public List<Volume> getVolumes() {
     public JobSpec asKubernetes() {
         JobSpecBuilder jobSpecBuilder = new JobSpecBuilder();
         final PodTemplateSpecFluent.SpecNested<JobSpecFluent.TemplateNested<JobSpecBuilder>> dsl =
-                jobSpecBuilder.withNewTemplate().withNewSpec().withContainers(container).withRestartPolicy("Never").withVolumes(volumes);
+                jobSpecBuilder.withNewTemplate().withNewSpec().withContainers(container)
+                        .withRestartPolicy("Never").withVolumes(volumes).withServiceAccountName(serviceAccountName)
+                        .withNodeSelector(Map.of("iam.gke.io/gke-metadata-server-enabled", "true"));
```

Comment:
Perhaps we don't need this anymore?

Reply:
And if service account is optional, then should it not only be enabled when the service account is specified?